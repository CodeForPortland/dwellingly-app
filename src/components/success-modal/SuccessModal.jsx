import React from 'react';
import PropTypes from 'prop-types';
import './SuccessModal.scss';

class SuccessModal extends React.Component {
  render() {
    // Render nothing if the "show" prop is false
    if(!this.props.show) {
      return null;
    }

    return (
      <div className="full-window-modal" >
        <div className="modal-content">
          <p className="title">
            {this.props.children}
          </p>
          <div className="button-container">
            <button form='newPropertyForm' className="btn"
              id="yes-button" type="button"
              onClick={this.props.onClick}>
              Add Another
            </button>
            <button className="cancel" onClick={this.props.onClose}
              type="button">
              Return Home
            </button>
          </div>
        </div>
      </div>
    );
  }
}

SuccessModal.propTypes = {
  onClick: PropTypes.func.isRequired,
  onClose: PropTypes.func.isRequired,
  show: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired
};

export default SuccessModal;
