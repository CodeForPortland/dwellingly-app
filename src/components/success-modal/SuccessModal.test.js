import React from 'react';
import renderer from 'react-test-renderer';
import SuccessModal from './SuccessModal';

const SuccessModalTest = (props) =>
<SuccessModal
{...props}
/>;

describe('SuccessModal Component', () => {
  it('correctly renders SuccessModal component', () => {
    const SuccessModalComponent = renderer.create(SuccessModalTest).toJSON();
    expect(SuccessModalComponent).toMatchSnapshot();
  });
});
