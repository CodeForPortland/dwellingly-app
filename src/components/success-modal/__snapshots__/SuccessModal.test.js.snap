// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessModal Component correctly renders SuccessModal component 1`] = `null`;
